src/lib/import_export/binary/binary_writer.cpp

```diff
@@ -200,6 +204,33 @@ void BinaryWriter::_write_segment(const DictionarySegment<T>& dictionary_segment
                             *dictionary_segment.attribute_vector());
 }
 
+template <typename T>
+void BinaryWriter::_write_segment(const FixedStringDictionarySegment<T>& fixed_string_dictionary_segment,
+                                  std::ofstream& ofstream) {
+  Assert(fixed_string_dictionary_segment.compressed_vector_type(),
+         "Expected DictionarySegment to use vector compression for attribute vector");
+  Assert(is_fixed_size_byte_aligned(*fixed_string_dictionary_segment.compressed_vector_type()),
+         "Does only support fixed-size byte-aligned compressed attribute vectors.");
```

Comment:
This is also asserted in _compressed_vector_width. Maybe you could harmonize this across the entire file?